src/plan_expand_hypertable.c

```diff
@@ -1145,7 +1145,7 @@ ts_plan_expand_timebucket_annotate(PlannerInfo *root, RelOptInfo *rel)
 	/* Walk the tree and find restrictions or chunk exclusion functions */
 	timebucket_annotate_walker((Node *) root->parse->jointree, &ctx);
 
-	if (ctx.propagate_conditions != NIL)
+	if (ctx.propagate_conditions != NIL && ts_guc_enable_qual_propagation)
```

Comment:
It is easier to maintain this code if the check is put inside the `propagate_join_quals`. That way, if a call is added somewhere else, there is no risk that a join propagation is done for that case, which means easier debugging.